### PR TITLE
Remove duplicate code and reuse templating from upstream

### PR DESCRIPTION
**Pull request by fabianofranz**
Fixes https://github.com/openshift/origin/issues/10957
Fixes https://github.com/openshift/origin/issues/11035

We upstreamed CLI templating, so this removes the code we had in Origin and reuse it from upstream. What's left is required to allow the different positioning of "Usage" in our help.

PTAL @smarterclayton @juanvallejo

### CODE REVIEW

**Comment by fabianofranz** [comment]
also @deads2k

File: pkg/cmd/templates/templater.go [hunk 1]
```diff
@@ -33,10 +34,25 @@ func ActsAsRootCommand(cmd *cobra.Command, filters []string, groups ...CommandGr
 	return templater
 }
 
+func mainUsageTemplate() string {
```
Comment (juanvallejo) [inline pkg/cmd/templates/templater.go]: This looks pretty similar to [ktemplates.MainUsageTemplate](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/templates/templates.go#L77), any reason to re-define it on our codebase?

**Comment by juanvallejo** [comment]
one minor comment, otherwise lgtm

File: pkg/cmd/templates/templater.go (hunk 1 above)
Comment (fabianofranz) [inline pkg/cmd/templates/templater.go]: It's similar but the order is slightly different - note how `kubectl` has the usage at the bottom and we don't. Not sure if we want to converge at some point, but currently we don't.

**Comment by fabianofranz** [comment]
> one minor comment, otherwise lgtm

needs /lgtm on its own line if you think it's good to go  ;)

**Pull request merged**